### PR TITLE
Depend on Nook

### PR DESCRIPTION
**Pull request by lukechurch**

### CODE REVIEW

**Comment by lukechurch** [comment]
cc @ktkization 

I'd like the configurator to depend on Nook so that we can pull in the shared code for - e.g. handling pubsub and authentication, without duplicating it between the repos.

We may well end up merging the configurator repo into Nook in the longer term in any case.

Downgrading the version of firebase used in the configurator is needed to avoid a dependency conflict with Nook

File: webapp/pubspec.yaml [hunk 1]
```diff
@@ -5,11 +5,17 @@ environment:
...
-  firebase: ^7.2.0
+  firebase: ^5.0.0
   http: ^0.12.0
   intl: ^0.16.0
   chartjs:
     git: https://github.com/marianamarasoiu/chartjs.dart.git
+  nook:
+    git:
+      url: https://github.com/larksystems/nook.git
+      path: webapp
```
Comment (marianamarasoiu) [inline webapp/pubspec.yaml:16]: should this be `webapp/lib`?

**Review by marianamarasoiu — APPROVED** [review]
thanks @lukechurch , lgtm!

File: webapp/pubspec.yaml (hunk 1 above)
Comment (lukechurch) [reply webapp/pubspec.yaml:16]: I'll find out ;)

File: webapp/pubspec.yaml [hunk 2]
```diff
@@ -5,11 +5,17 @@ environment:
...
 dependencies:
-  firebase: ^7.2.0
+  firebase: ^5.0.0
   http: ^0.12.0
   intl: ^0.16.0
   chartjs:
     git: https://github.com/marianamarasoiu/chartjs.dart.git
+  nook:
+    git:
+      url: https://github.com/larksystems/nook.git
```
Comment (danrubel) [inline webapp/pubspec.yaml:15]: I always use a `git` url as in `git@github.com:larksystems/nook.git`
but that's out of habbit... I don't know which one is better or more common.

**Comment by danrubel** [comment]
LGTM!

**Pull request merged**